wp-includes/class-wp-offline-page.php

```diff
@@ -0,0 +1,157 @@
+<?php
+/**
+ * WP_Offline_Page class.
+ *
+ * @package PWA
+ */
+
+/**
+ * WP_Offline_Page class.
+ * Much of this is taken from wp-admin/privacy.php.
+ */
+class WP_Offline_Page {
+
+	/**
+	 * The option group for the offline page option.
+	 *
+	 * @var string
+	 */
+	const OPTION_GROUP = 'reading';
+
+	/**
+	 * The option name.
+	 *
+	 * @var string
+	 */
+	const OPTION_NAME = 'wp_offline_page';
+
+	/**
+	 * The ID of the settings section.
+	 *
+	 * @var string
+	 */
+	const SETTING_ID = 'pwa_offline_page';
+
+	/**
+	 * Inits the class.
+	 */
+	public function init() {
+		add_action( 'admin_init', array( $this, 'register_setting' ) );
+		add_action( 'admin_init', array( $this, 'settings_field' ) );
+	}
+
+	/**
+	 * Registers the offline page setting.
+	 */
+	public function register_setting() {
+		register_setting(
+			self::OPTION_GROUP,
+			self::OPTION_NAME,
+			array(
+				'type'              => 'integer',
+				'sanitize_callback' => array( $this, 'sanitize_callback' ),
+			)
+		);
+	}
+
+	/**
+	 * Sanitize callback for the setting.
+	 * Mainly taken from wp-admin/privacy.php.
+	 *
+	 * @todo: Ensure this is stored with a custom post status.
```

Comment:
For background, please see @westonruter's [https://github.com/xwp/pwa-wp/issues/9#issuecomment-403262531](https://github.com/xwp/pwa-wp/issues/9#issuecomment-403262531).